Added validation on make setup target

Nr18 opened the pull request:
Added validation to the setup target and raise a clear error when pyenv is not found on your system.

*Issue #, if available:*

*Description of changes:*
Added a check if pyenv could be found on the developers system, if not display a better error describing the problem.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

sanathkr commented on Makefile:
```diff
@@ -1,10 +1,17 @@
+
+setup: install
+ifndef PYENV
+    $(error "pyenv was not found, please make sure that pyenv is accessible by adding it to your PATH variable in your bash_profile.")
```
Can we generalize this a bit by saying: `make sure pyenv is accessible in your path, (usually by adding to PATH variable in bash_profile, zshrc, or other locations based on your platform)`

I don't want someone to think bash_profile is a requirement when they are using a Z-Shell or Windows

jfuss replied:
Would it be worth add the url for pyenv here as well? So that way, customers can jump right to the pyenv install guide?

Nr18 replied:
I will change it and add a link to the docs, good feedback

jfuss commented on Makefile:
```diff
@@ -1,10 +1,17 @@
+	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer 2> /dev/null | bash
+
+setup: install
+ifndef PYENV
```
Will this exit the command at all? If not should we be? We know that it isn't installed and therefore can't use pyenv at the moment.

Nr18 replied:
Interesting point now it will install it on the first attempt and then report that it's not configured. I was working on a implementation to get installed properly on my mac but that makes the Makefile complex. Shall i remove the installation and just message that pyenv should be installed properly before you can use it?

Nr18 commented:
So i have removed the installation part from the Makefile and added a note to the `DEVELOPMENT_GUIDE.rst`.

From my Macbook the environment didn't got activated and i needed to place it in one shell in order to get it working:

```$(shell eval "$$(pyenv init -)" && eval "$$(pyenv virtualenv-init -)" && pyenv activate $(NAME) && pyenv local $(NAME))```

Do you have any concerns that this will not work on another device/os?

Nr18 commented:
@jfuss regarding the change from master to develop, i used master due to the following in `CONTRIBUTING.md`

```1. You are working against the latest source on the *master* branch.```

Would you like me to update that to develop?

jfuss commented:
> Would you like me to update that to develop?

That would be awesome.

> Do you have any concerns that this will not work on another device/os?

For me, no but my shell scripting/makefile abilities are very limited. I will be leaning on @sanathkr on this one.

Nr18 commented:
Branch name has been updated, @sanathkr do you see any concerns regarding my shell activation changes?

jfuss commented on Makefile:
```diff
@@ -1,10 +1,42 @@
+	$(shell eval "$$(pyenv init -)" && eval "$$(pyenv virtualenv-init -)" && pyenv activate $(NAME) && pyenv local $(NAME))
+	
+# Remove all pyenv versions to start fresh	
+clean:
```
Can we add pyenv to this command? I can imagine a clean command that cleans all .pyc files or tox. It will make this command a little clearer on what will happen.

Nr18 replied:
Not sure how to clean tox things but added the pyc and python cache folders

jfuss commented on Makefile:
```diff
@@ -1,10 +1,42 @@
+	
+# Remove all pyenv versions to start fresh	
+clean:
+	rm -rf ~/.pyenv/versions/*
```
This is a little scary to me. What is the reason we need to -f this?

Nr18 replied:
I removed it but it will ask for a are you sure confirmation on: /.pyenv/versions/2.7.14/lib/libpython2.7.a?

Nr18 commented:
@jfuss i have added some documentation in the Makefile that will be displayed when you just enter `make` (copied the idea from the https://github.com/aws-samples/cookiecutter-aws-sam-python repo)

I also included the install target again for those who want to use it.

jfuss commented:
@Nr18 This looks like it was mostly taken from the cookiecutter project right? Generally, this looks good to me.

@heitorlessa Would you mind taking a look at this PR? I am not a makefile magician like you :)

Nr18 commented:
@jfuss yes

sanathkr commented on Makefile:
```diff
@@ -1,25 +1,80 @@
+	$(info [*] Activate virtualenv $(NAME)...)
+	$(shell eval "$$(pyenv init -)" && eval "$$(pyenv virtualenv-init -)" && pyenv activate $(NAME) && pyenv local $(NAME))
+	
+clean:
```
I know this is convenient to clean things, but can we actually remove it from here? This might work for someone setting up SAM Translator with pyenv and other things, but they are by no means required to. I am concerned that we are going beyond the scope.

Thoughts?

Nr18 commented:
So i love having a way to clean everything and start fresh but i have removed it so that the PR can go forward.

sanathkr commented:
Appreciate making the changes @Nr18. Looks good to me.

sanathkr commented:
@jfuss can you take a quick look and approve?

jfuss commented:
Looks good to me. 

@Nr18, I would agree but I think we need to think about what that actually means. Do we include nuking everything or just cleaning .pyc files, etc. When using pyevn nuking is pretty easy and I think I would rather have nuking be explicit by the dev than something that could easily go wrong. A deeper discussion is worth having to see what we can/ should do here though.

jfuss commented:
@sanathkr can you merge and to the conventional commit? I’m on mobile which makes this somewhat difficult.
